File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        for (int i = 0; i < nums.length-1;i++){
            int compare = nums[i];
            for (int j = i+1; j < nums.length; j++){
                if (compare == nums[j])
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/permutations/submission-1.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []
        cur = []
        choices = []

        def dfs(choices):
            if not choices:
                res.append(cur.copy())
                return
            for i, num in enumerate(choices):
                temp = choices.copy()
                cur.append(num)
                temp.pop(i)
                dfs(temp)

                cur.pop()

        dfs(nums)
        return res